spec/requests/custom_actions_spec.rb

```diff
@@ -128,7 +128,7 @@ def expect_result_to_have_custom_actions_hash
 
       expect_result_to_have_keys(%w(id href actions))
       action_specs = response.parsed_body["actions"]
-      expect(action_specs.size).to eq(1)
+      expect(action_specs.size).to eq(3)
```

Comment:
It looks like this behavior changed during a refactoring. Nit, but it might help to break up this commit into the refactoring part and the part that introduced the change in behavior.